### DISCUSSION TITLE
Teleport: breaks after re-render component

### DISCUSSION THREAD

**Discussion by robsontenorio** [livewire/livewire#6954] [7. Bugs (archived)]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

After component re-render when clicking save, the teleport breaks.

```php
<?php

use Livewire\Volt\Component;

new class extends Component 
{
    public function save()
    {
    }
}; ?>

<div>

    <div>
        Hello!

        @teleport('#target')
            <span>Hey!</span>
        @endteleport
    </div>

    It will desapear: <span id="target"></span>

    <div>
        <button wire:click="save">Save</button>
    </div>
</div>
```

But if place a **random** `wire:key` it works. **What I am missing?**

```html
    <div wire:key="{{ random() }}">
        Hello!

        @teleport('#target')
           <span>Hey!</span>
        @endteleport
    </div>
```

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by Sparclex** [comment]
Seem like it is not supported to teleport within the same component as stated in the documentation: 
https://livewire.laravel.com/docs/teleport
> Livewire only supports teleporting HTML outside your components. For example, teleporting a modal to the `<body>` tag is fine, but teleporting it to another element within your component will not work.
